pycbc/waveform/bank.py

```diff
@@ -394,24 +403,157 @@ def approximant(self, index):
     def __len__(self):
         return len(self.table)
 
-    def template_thinning(self, injection_parameters, threshold):
-        from pycbc.pnutils import mass1_mass2_to_tau0_tau3
+    def template_thinning(self, gwstrain):
+        if gwstrain.injcutter is None or \
+                gwstrain.injcutter['chirp_time_threshold'] is None:
```

Comment:
So I guess if we make injcutter a separate instance this is then:

```
template_thinning(self, injcutter):
```

I think this is the right way to do it, rather than having specific key-word arguments for all of the injcutter parameters, both here and in the second function. Ie. pycbc_inspiral should not be worrying about the contents of the injcutter instance, it should just pass it along. There is already too much code/command-line options in pycbc_inspiral, which makes it hard to maintain or to write things like pycbc_inspiral_precession and pycbc_inspiral_coherent.

However, if there is a potential for using this function outside of pycbc_inspiral, then maybe it makes sense to have two functions, one which takes the chirp_time_threshold and other options explicitly and the second which just take injcutter and then calls from that to the second function.